File: ios/JsiPreview.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(JsiPreview, NSObject)

+ (BOOL)requiresMainQueueSetup {
    return false;
}

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/JsiPreview.swift
@objc(JsiPreview)
class JsiPreview: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        for _ in 0...1000000 {}
        
        resolve(a*b)
    }
}
